src/rebar_app_discover.erl

```diff
@@ -48,7 +48,9 @@ merge_deps(AppInfo, State) ->
     State1 = lists:foldl(fun(Profile, StateAcc) ->
                                  AppProfDeps = rebar_state:get(AppState, {deps, Profile}, []),
                                  TopLevelProfDeps = rebar_state:get(StateAcc, {deps, Profile}, []),
-                                 ProfDeps2 = lists:keymerge(1, TopLevelProfDeps, AppProfDeps),
+                                 ProfDeps2 = dedup(lists:keymerge(1,
```

Comment:
ukeymerge wouldn't work because duplicate deps with different sources would be crushed silently. umerge alone wouldn't necessarily work as it expects lists to be free of duplicates to begin with. So we want a single keymerged list with duplicate elements removed. Not sure there's another way to do exactly that.